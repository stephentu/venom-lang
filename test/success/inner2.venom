def foo(x::int) -> int =
  def foo(x::int) -> int = x; end
  foo(x);
end
print(foo(30));

def bar{A}(a::A) -> A =
  def bar{B}(a::B) -> B = a; end
  bar{A}(a);
end
print(bar{int}(40));
print(bar{string}("hello"));
