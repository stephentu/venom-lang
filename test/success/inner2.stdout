30
40
hello
